### PR TITLE
[ACS][CallAutomation] Live Test Updated V2

### PR DESCRIPTION
**Pull request by minwoolee-msft**
Live Test Updated for CallAutomation + Recorded 
- LocalAuth disabled for ServiceBus
- PhoneNumber test now randomized to avoid duplicated tests
- Various Bug Fixes
- more redaction added

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAutomatedLiveTestBase.java [hunk 1]
```diff
@@ -278,11 +281,8 @@ protected String parseIdsFromIdentifier(CommunicationIdentifier communicationIde
             : removeAllNonChar(communicationIdentifierModel.getRawId());
     }
 
-    /* Change the plus + sign to it's unicode without the special characters i.e. u002B.
-     * It's required because the dispatcher app receives the incoming call context for PSTN calls
-     * with the + as unicode in it and builds the topic id with it to send the event.*/
     protected static String removeAllNonChar(String input) {
-        return input.replace("+", "u002B").replaceAll("[^a-zA-Z0-9_-]", "");
+        return input.replaceAll("[^a-zA-Z0-9_-]", "");
```
Comment (huachuandeng) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAutomatedLiveTestBase.java:285]: Why the first replace is not needed any more?

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationLiveTestBase.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ public class CallAutomationLiveTestBase extends TestProxyTestBase {
     protected static final String PHONE_USER_1
         = Configuration.getGlobalConfiguration().get("TARGET_PHONE_NUMBER", "+16471234567");
     protected static final String MEDIA_SOURCE
-        = Configuration.getGlobalConfiguration().get("ACS_MEDIA_SOURCE", "https://contoso.com/music.wav");
+        = Configuration.getGlobalConfiguration().get("FILE_SOURCE_URL", "https://contoso.com/music.wav");
```
Comment (huachuandeng) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationLiveTestBase.java:72]: Why do we need to change this?

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAutomatedLiveTestBase.java (hunk 1 above)
Comment (minwoolee-msft) [reply sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAutomatedLiveTestBase.java:285]: this was a bug in dispatcher that was addressed by redployment. This special treatment is now removed as dispatcher correctly parses MRI

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationLiveTestBase.java (hunk 2 above)
Comment (minwoolee-msft) [reply sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationLiveTestBase.java:72]: This matches with the same format as other 3 SDKs. All 3 SDK uses "FILE_SOURCE_URL" as file source, but Java for some whatever reason uses this different name. This is to uniform as much as possible between SDK live test to reduce pain to have different treatment of each SDK live tests

**Review by huachuandeng — APPROVED** [review]

**Pull request merged**